src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/RealExternalServiceDAO.scala

```diff
@@ -0,0 +1,43 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import org.broadinstitute.dsde.rawls.model.AgoraEntity
+import org.broadinstitute.dsde.rawls.model.JobStatus
+import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+import spray.client.pipelining._
+import spray.http._
+import spray.http.HttpHeaders.Cookie
+import spray.httpx.SprayJsonSupport._
+import spray.json._
+
+/**
+ * @author tsharpe
+ */
+class RealExternalServiceDAO( methodRepoConf: Config, execSvcConf: Config)( implicit system: ActorSystem ) extends ExternalServiceDAO {
+
+  override def getMethodConfig( namespace: String, name: String, version: String, authCookie: HttpCookie ): Try[HttpResponse] = {
+    val url = methodRepoConf.getString("getConfigPath").format(namespace,name,version)
```

Comment:
we should not be passing in Config object to the constructor of this class. We should pass in exactly what is required and in the required format. I think in this case we should pass in the root of the url, not a url pattern that we later format. I think we gain little from externalizing the full format and lose readability. In all likelihood any changes to the request will require code changes anyway so why bother putting it somewhere else?